### PR TITLE
feat(splitter): implement indigo splitter changes

### PR DESCRIPTION
**Pull request by didimmova**
### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/splitter/splitter.component.ts [hunk 1]
```diff
@@ -175,6 +175,19 @@ export class IgxSplitterComponent implements AfterContentInit {
...
+     * True by default
+     * 
+     * @example
+     * ```html
+     * <igx-splitter [showControls]='false''>
+     * </igx-splitter>
+     * ```
+     */
+    @Input()
+    public showControls: boolean = true; // Input to toggle showing/hiding controls
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/splitter/splitter.component.ts:189]: I am not so sure about the name. Perhaps something like `collapsible` can be a better naming as it more widely captures what is being disabled/enabled.

File: projects/igniteui-angular/src/lib/splitter/splitter.component.ts [hunk 2]
```diff
@@ -366,6 +379,13 @@ export class IgxSplitBarComponent {
     @HostBinding('class.igx-splitter-bar-host')
     public cssClass = 'igx-splitter-bar-host';
 
+     /**
+     * Sets the visibility of the handle and expanders in the splitter bar.
+     */
+    @Input()
+    public showControls: boolean;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/splitter/splitter.component.ts:386]: Didn't you declare this already on line 189?

File: projects/igniteui-angular/src/lib/splitter/splitter-bar.component.html [hunk 3]
```diff
@@ -7,8 +7,9 @@
     (dragStart)="onDragStart($event)"
     (dragMove)="onDragMove($event)"
     (dragEnd)="onDragEnd($event)"
+    [class.show-controls]="showControls"
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/splitter/splitter-bar.component.html:10]: Let's keep the BEM naming convention here.

**Review by simeonoff — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss [hunk 4]
```diff
@@ -78,9 +79,15 @@
     $hitbox-debug-color: rgba(coral, .24);
 
     $variant: map.get($theme, '_meta', 'variant');
-    $size: map-get($theme, 'size');
-    $splitter-size: ($size / ($size * 0 + 1) + 2);
-    $slim-splitter: calc( 1 / $splitter-size);
+
+    //remove unit from size
+    $size: math.div(map-get($theme, 'size'), map-get($theme, 'size') * 0 + 1);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss:84]: Why do you multiply the size * 0 and then add 1 to it just to arrive at size = size / 1 which is simply the value of size? s / s * 0 * 1 = s?

**Review by simeonoff — COMMENTED** [review]
Check math.

File: projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss [hunk 5]
```diff
@@ -79,6 +80,15 @@
 
     $variant: map.get($theme, '_meta', 'variant');
 
+    //remove unit from size
+    $size: math.div(map-get($theme, 'size'), map-get($theme, 'size') * 0 + 1);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss:84]: Don't use old syntax like `map-get`. You already have the module `map` module imported.

```suggestion
    $size: math.div(map-get($theme, 'size'), map-get($theme, 'size') * 0 + 1);
```

File: projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss (hunk 4 above)
Comment (didimmova) [reply projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss:84]: Yeah, I was amazed myself but obviously this is the way to remove the unit of the variable :D Dividing it just by 1 doesn't work. I don't know why, honostly

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/splitter/_splitter-theme.scss]: Ok, got it. Can you make this as a function in the theming library, under the utils/math module?

**Review by simeonoff — APPROVED** [review]

**Pull request merged**